Gemfile

```diff
@@ -43,5 +43,12 @@ group :development, :test do
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'rspec-rails'
+
+  gem 'devise', '~> 3.4.1'
+  gem 'bootstrap-sass', '~> 3.3.1.0'
+  gem 'devise-bootstrap-views', '~> 0.0.3'
+  gem 'execjs'
+  gem 'therubyracer'
```

Comment:
You had added this gems to 
```ruby
group :development, :test do
```
It means that they should be used ONLY in test and development environments. But that is not true.
Please, move this 5 lines to the upper section of Gemfile